runtime/src/lib.rs

```diff
@@ -284,6 +284,49 @@ impl pallet_utility::Config for Runtime {
     type WeightInfo = pallet_utility::weights::SubstrateWeight<Runtime>;
 }
 
+parameter_types! {
+    pub const DisallowPermissionlessEnterDuration: BlockNumber = 0;
+    pub const DisallowPermissionlessExtendDuration: BlockNumber = 0;
+
+    pub const RootEnterDuration: BlockNumber = 5 * 60 * 3; // 3 hours
+    pub const RootExtendDuration: BlockNumber = 5 * 60 * 3; // 3 hours
+
+    pub const DisallowPermissionlessEntering: Option<Balance> = None;
+    pub const DisallowPermissionlessExtending: Option<Balance> = None;
+    pub const DisallowPermissionlessRelease: Option<BlockNumber> = None;
+}
+
+pub struct SafeModeWhitelistedCalls;
+impl Contains<RuntimeCall> for SafeModeWhitelistedCalls {
+    fn contains(call: &RuntimeCall) -> bool {
+        matches!(
+            call,
+            RuntimeCall::Sudo(_)
+                | RuntimeCall::System(_)
+                | RuntimeCall::SafeMode(_)
+                | RuntimeCall::Timestamp(_)
+        )
+    }
+}
```

Comment:
just wanted to double check we don't want to whitelist any other calls, and that we can perform a runtime upgrade when in safemode as that seems a likely safemode activity

Reply:
I think this is a safe default, the calls here are basic operations critical for the blockchain, and as such runtime upgrades can still be performed. No other pallets that are irrelevant are included here.